sound/soc/codecs/rt1308-sdw.c

```diff
@@ -118,7 +118,7 @@ static int rt1308_clock_config(struct device *dev)
 static int rt1308_read_prop(struct sdw_slave *slave)
 {
 	struct sdw_slave_prop *prop = &slave->prop;
-	int nval, i, num_of_ports = 1;
```

Comment:
What does the "1" stand for? DP0? I don't get why we don't need to count it.

Reply:
no idea. If DP0 was used then it should be properly initialized.

Reply:
After looking into the code, I am pretty sure the 1 is for DP0 and we need it now. The num_of_ports is used to allocate slave->port_ready and we use port_ready with [&slave->port_ready[port]](https://github.com/thesofproject/linux/blob/topic/sof-dev/drivers/soundwire/bus.c#L1216). So, it will be a problem if we try to complete the latest port if we don't add 1 here. For example, rt1308 has one port which is port 1. If we allocate only 1 port_ready for rt1308, it will be a problem when we try to complete port_ready[1].
I think we will meet the same issue with [num_of_ports = hweight32(nval) + dp0; /* add DP0 */](https://github.com/thesofproject/linux/blob/topic/sof-dev/drivers/soundwire/mipi_disco.c#L388) if ``dp0 = 0``. 
Also, some codec may not use sequencing port number. For example, rt711's port numbers are 2, 3, 4. I think it should be a problem when a port 4 interrupt is triggered even with the existing code.

Reply:
@bardliao we don't need support for DP0/BRA for now. This will only be needed when external VAD is supported, and for now we don't have a good timeline.
So yes *WHEN* DP0 is supported, we will need to increase the port count. For now it's not correct to allocate memory for no good reason.

Reply:
@bardliao correction to my previous comment:

The entire use of slave->port_ready[] is incorrect in the upstream code. You are absolutely right that the ports need not be contiguous, and on top of this the same port can be sink or source, so should not be counted twice.

I still believe my patch is correct, it just counts the ports that are described and used. We however need to fix the entire port_ready[] use with a separate PR. The good thing is that for now no codec uses the port_ready interrupt (they all use the simplified state machine), so there is no effect. It's still wrong and needs to be fixed.

Reply:
> @bardliao correction to my previous comment:
> 
> The entire use of slave->port_ready[] is incorrect in the upstream code. You are absolutely right that the ports need not be contiguous, and on top of this the same port can be sink or source, so should not be counted twice.
> 
> I still believe my patch is correct, it just counts the ports that are described and used. We however need to fix the entire port_ready[] use with a separate PR. The good thing is that for now no codec uses the port_ready interrupt (they all use the simplified state machine), so there is no effect. It's still wrong and needs to be fixed.

Agree. we need a separate patch to fix the slave->port_ready[] issue.